src/ADAL.PCL.CoreCLR/ClientAssertionCertificate.cs

```diff
@@ -0,0 +1,107 @@
+﻿//----------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+using System.Globalization;
+using System.Security.Cryptography.X509Certificates;
+
+namespace Microsoft.IdentityModel.Clients.ActiveDirectory
+{
+    /// <summary>
+    /// Containing certificate used to create client assertion.
+    /// </summary>
+    public sealed class ClientAssertionCertificate : IClientAssertionCertificate
```

Comment:
We need must use a copy of this file because the way to access private/public keys from X509 certificates is different between NetFx and CoreFx.
Furthermore we cannot extract these differences into a platform interface (say `ICryptographHelper`) because that would require using types (`AsymmetricSecurityKey`, `X509Certificate2`) on an interface in the _ADAL.PCL_ library. These types are _not_ available in .NET Standard 1.1. Chaning the platform-agnostic assembly to target .NET Standard 1.3 would mean removing the ability to run against .NET Framework 4.5.